integration/linux/env/vars.sh

```diff
@@ -140,49 +140,33 @@ TBBROOT=$(get_script_path "${vars_script_name:-}")/..
 
 TBB_TARGET_ARCH="intel64"
 
-if [ ! -z "$SETVARS_ARGS" ] && [ "$SETVARS_CALL" "==" "1" ]; then
-    for arg in "$SETVARS_ARGS"; do
-        case "$arg" in
-        (intel64|ia32)
-            TBB_TARGET_ARCH="${arg}"
-            ;;
-        (*) ;;
-        esac
-    done
-elif [ ! -z "$SETVARS_ARGS" ]; then
-        for arg in "$SETVARS_ARGS"; do
-        case "$arg" in
-            (intel64|ia32)
-            TBB_TARGET_ARCH="${arg}"
-            ;;
-            (*)
-            >&2 echo "ERROR: Unknown argument $arg"
-            return 255 2>/dev/null || exit 255
-            ;;
-        esac
-    done
+if [ -n "${SETVARS_ARGS:-}" ]; then
+  tbb_arg_ia32="$(expr "${SETVARS_ARGS:-}" : '^.*[:space:]*\(ia32\)[:space:]*')"
```

Comment:
Now that I look at this a second time, there's no value, in this case, to include the `[:space:]*` expressions, it's not providing more precision. So it would be simpler to just use the following regex:
```
'^.*\(ia32\)'
```
until we can define a more precise way of doing this in a POSIX conformant way.

Reply:
Is there any need in `()`? What if just `'^.*ia32'`?

Reply:
Never mind, `()` are needed. From `man expr`:
> Pattern matches return the string matched between \\( and \\) or null; if \\( and \\) are not used, they return the number of characters matched or 0.

Reply:
Applied.